### PR TITLE
CLI enhancements

### PR DESCRIPTION
**Pull request by marcauberer**
- Allow `--lang auto` to determine the language, based on the file extension
- Print error message when the cli is unable to find compiler executable
- Throw exception if only one of `COMMENT_BLOCK_IDEN_OPEN` and `COMMENT_BLOCK_IDEN_CLOSE` is specified
- Rename CommentChars to CommentIden
- Refactoring
- README fixes

### CODE REVIEW

File: cli/ccom.go [hunk 1]
```diff
@@ -33,13 +33,13 @@ func main() {
 				Usage:   "Benchmark the specified compiler executable and run it n times",
 			},
 			&cli.StringFlag{
-				Name:    "block-comment-chars-open",
-				Aliases: []string{"bcco"},
+				Name:    "block-comment-iden-open",
+				Aliases: []string{"bcio"},
 				Usage:   "Specifies the opening block comment character(s) of your data format",
```
Comment (Splines) [inline cli/ccom.go:38]: change to `identifier`s (note the multiple occurrences in this file)

File: cli/preprocessor.go [hunk 2]
```diff
@@ -95,13 +96,16 @@ func analyze(
...
 	}
 	// Ensure value of comment char
-	if *lineCommentChars == "" && *blockCommentCharsOpen == "" && *blockCommentCharsClose == "" {
-		*lineCommentChars = "#"
-		*blockCommentCharsOpen = ""
-		*blockCommentCharsClose = ""
+	if *lineCommentIden == "" && *blockCommentIdenOpen == "" && *blockCommentIdenClose == "" {
+		*lineCommentIden = "#"
+		*blockCommentIdenOpen = ""
+		*blockCommentIdenClose = ""
```
Comment (Splines) [inline cli/preprocessor.go:105]: ? shouldn't this be an error as well?

Comment (Splines) [reply cli/preprocessor.go:105]: https://github.com/compose-generator/ccom/blob/86f126e1e83fb945f25e0d54dcf0dbe6076bcaec/compiler/java/src/main/java/com/c/lexer/Lexer.java#L116-L122

I implemented this as an error

**Review by Splines — CHANGES_REQUESTED** [review]

File: cli/ccom.go (hunk 1 above)
Comment (marcauberer) [reply cli/ccom.go:38]: Don't now what you mean, sorry. What's wrong with that?

File: cli/preprocessor.go (hunk 2 above)
Comment (marcauberer) [reply cli/preprocessor.go:105]: Yes, you are right. Thats another remnant here ;)
Will fix that quickly ...

File: cli/ccom.go (hunk 1 above)
Comment (marcauberer) [reply cli/ccom.go:38]: `iden` -> `identifier`?

Comment (Splines) [reply cli/ccom.go:38]: I mean the value for the `Usage` key: "Specifies the opening block comment character(s) of your data format"
You've replaced "chars" by "iden" (line 36+37), so why not replace "characters" by "iden" or "identifier" as well...

File: cli/preprocessor.go [hunk 3]
```diff
@@ -119,35 +121,51 @@ func runCompilerExecutable(
...
+	}
+	// Execute compiler with user inputs
+	return util.ExecuteAndWaitWithOutput(executableName, strconv.FormatBool(modeSingle), fileInput, dataInput, lineCommentIden, blockCommentIdenOpen, blockCommentIdenClose)
 }
 
-func getCommentCharsFromLang(lang string) (lineCommentChars string, blockCommentCharsOpen string, blockCommentCharsClose string) {
+func getCommentIdenFromLang(lang string, fileInput string) (lineCommentIden string, blockCommentIdenOpen string, blockCommentIdenClose string) {
+	if lang == "auto" {
+		if !util.FileExists(fileInput) {
+			log.Fatal("Please use lang 'auto' only in combination of valid file paths as file input")
```
Comment (Splines) [inline cli/preprocessor.go:149]: Does `log.Fatal(...)` terminate or is it just writing to the log?

Comment (marcauberer) [reply cli/preprocessor.go:149]: `log.Fatal` termintates.

File: cli/ccom.go (hunk 1 above)
Comment (marcauberer) [reply cli/ccom.go:38]: Ahh okay. Know what you mean. Gonna fix that real quick ...

**Review by Splines — APPROVED** [review]

File: cli/ccom.go (hunk 1 above)
Comment (marcauberer) [reply cli/ccom.go]: Should be fine now.

**Pull request merged**